src/feature/dirauth/process_descs.c

```diff
@@ -121,6 +128,40 @@ add_fingerprint_to_dir(const char *fp, authdir_config_t *list,
   return;
 }
 
+/** Add the ed25519 key <b>edkey</b> to the smartlist of fingerprint_entry_t's
+ * <b>list</b>, or-ing the currently set status flags with
+ * <b>add_status</b>.
+ */
+/* static */ void
+add_ed25519_to_dir(const char *edkey, authdir_config_t *list,
+                   router_status_t add_status)
+{
+  char *ed25519_key;
+  char d[DIGEST_LEN];
```

Comment:
d is 160 bits long, but digest256_from_base64() expects a 256-bit array.

A unit test would have discovered this memory safety issue.